src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -1640,6 +1643,26 @@ ep_rt_mono_init (void)
 	mono_profiler_set_thread_stopped_callback (profiler, profiler_eventpipe_thread_exited);
 }
 
+void
+ep_rt_mono_init_finish (void)
+{
+	if (mono_runtime_get_no_exec ())
+		return;
+
+	// Managed init of diagnostics classes, like registration of RuntimeEventSource (if available).
+	ERROR_DECL (error);
+
+	MonoClass *runtime_event_source = mono_class_from_name_checked (mono_defaults.corlib, "System.Diagnostics.Tracing", "RuntimeEventSource", error);
+	if (is_ok (error) && runtime_event_source) {
```

Comment:
this can fail if the type is removed

Reply:
yes and that is expected, then it won't just be called.